alts/src/main/java/io/grpc/alts/AltsChannelBuilder.java

```diff
@@ -103,47 +82,16 @@ protected NettyChannelBuilder delegate() {
 
   @Override
   public ManagedChannel build() {
-    if (!CheckGcpEnvironment.isOnGcp()) {
-      if (enableUntrustedAlts) {
-        logger.log(
-            Level.WARNING,
-            "Untrusted ALTS mode is enabled and we cannot guarantee the trustworthiness of the "
-                + "ALTS handshaker service");
-      } else {
-        Status status =
-            Status.INTERNAL.withDescription("ALTS is only allowed to run on Google Cloud Platform");
-        delegate().intercept(new FailingClientInterceptor(status));
-      }
-    }
     InternalNettyChannelBuilder.setProtocolNegotiatorFactory(
         delegate(),
-        new ClientAltsProtocolNegotiatorFactory(
-            targetServiceAccountsBuilder.build(), handshakerChannelPool));
+        credentialsBuilder.buildProtocolNegotiatorFactory());
```

Comment:
With this change (when `!CheckGcpEnvironment.isOnGcp()` and  `!enableUntrustedAlts`) we use a `FailingProtocolNegotiator` instead of the previous `FailingClientInterceptor` which fails each client call which is different behavior. May be the difference is not important but we might be able to keep the same behavior by throwing an exception from `credentialsBuilder.buildProtocolNegotiatorFactory()` and catching that here to apply `FailingClientInterceptor` on the `delegate()`.

Reply:
> May be the difference is not important

It doesn't seem very important. And there's really no reason it should be an interceptor. It is really the negotiator that is failing.

> but we might be able to keep the same behavior by throwing an exception from credentialsBuilder.buildProtocolNegotiatorFactory() and catching that here to apply FailingClientInterceptor on the delegate().

That would produce different code paths between the channel builder and credential. Since there's no gain to my knowledge to using the interceptor, it's not clear why we'd choose that approach.

Reply:
> It doesn't seem very important. And there's really no reason it should be an interceptor. It is really the negotiator that is failing.

Agreed and thought this change is more desirable and might be by design, but thought this is worth pointing out in case any client code depended on the old behavior.

> 
> > but we might be able to keep the same behavior by throwing an exception from credentialsBuilder.buildProtocolNegotiatorFactory() and catching that here to apply FailingClientInterceptor on the delegate().
> 
> That would produce different code paths between the channel builder and credential. Since there's no gain to my knowledge to using the interceptor, it's not clear why we'd choose that approach.

Okay